common/lib/xmodule/xmodule/modulestore/search.py

```diff
@@ -104,15 +105,25 @@ def find_path_to_course():
         # (e.g. sequential and videosequence) currently deal with this form of
         # representing nested positions. This needs to happen before jumping to a
         # module nested in more than one positional module will work.
+
         if n > 3:
             position_list = []
+            masquerading_student = False
+            if request and request.session.get('masquerade_settings') and \
```

Comment:
Why do we need to check if we have a request? why the `request` has default argument of `None`?

Reply:
`masquerading_as_student = masquerade_settings and masquerade_settings[usage_key.course_key].role == 'student'`

Reply:
because other flows dont have request object coming in for this class, some test that were failing because of test can be seen here in this build of test pipeline.
https://build.testeng.edx.org/job/edx-platform-python3-python-pipeline-pr/761/